src/extensions/default/Phoenix-Markdown/main.js

```diff
@@ -360,6 +372,187 @@ define(function (require, exports, module) {
+
+    CommandManager.register(Strings.HINT_H1, H1_COMMAND_ID, Handler.h1);
+    CommandManager.register(Strings.HINT_H2, H2_COMMAND_ID, Handler.h2);
+    CommandManager.register(Strings.HINT_H3, H3_COMMAND_ID, Handler.h3);
+    CommandManager.register(Strings.HINT_H4, H4_COMMAND_ID, Handler.h4);
+    CommandManager.register(Strings.HINT_BOLD, BOLD_COMMAND_ID, Handler.bold);
+    CommandManager.register(Strings.HINT_ITALIC, ITALIC_COMMAND_ID, Handler.italic);
+    CommandManager.register(Strings.HINT_STRIKE, STRIKE_COMMAND_ID, Handler.strikethrough);
+    CommandManager.register(Strings.HINT_CODE, CODE_COMMAND_ID, Handler.code);
+    CommandManager.register(Strings.HINT_BULLET, BULLET_COMMAND_ID, Handler.bullet);
+    CommandManager.register(Strings.HINT_NUMBERED, NUMBERED_COMMAND_ID, Handler.numbered);
+    CommandManager.register(Strings.HINT_QUOTE, QUOTE_COMMAND_ID, Handler.quote);
+    CommandManager.register(Strings.HINT_CODEBLOCK, CODEBLOCK_COMMAND_ID, Handler.codeblock);
+    CommandManager.register(Strings.HINT_PARAGRAPH, PARAGRAPH_COMMAND_ID, Handler.paragraph);
+    CommandManager.register(Strings.HINT_REFLOW, REFLOW_COMMAND_ID, Handler.reflow);
+
+    KeyBindingManager.addBinding(H1_COMMAND_ID, KeyboardPrefs.heading1);
+    KeyBindingManager.addBinding(H2_COMMAND_ID, KeyboardPrefs.heading2);
+    KeyBindingManager.addBinding(H3_COMMAND_ID, KeyboardPrefs.heading3);
+    KeyBindingManager.addBinding(H4_COMMAND_ID, KeyboardPrefs.heading4);
+    KeyBindingManager.addBinding(BOLD_COMMAND_ID, KeyboardPrefs.bold);
+    KeyBindingManager.addBinding(ITALIC_COMMAND_ID, KeyboardPrefs.italic);
+    KeyBindingManager.addBinding(STRIKE_COMMAND_ID, KeyboardPrefs.strikethrough);
+    KeyBindingManager.addBinding(CODE_COMMAND_ID, KeyboardPrefs.code);
+    KeyBindingManager.addBinding(BULLET_COMMAND_ID, KeyboardPrefs.bullet);
+    KeyBindingManager.addBinding(NUMBERED_COMMAND_ID, KeyboardPrefs.numbered);
+    KeyBindingManager.addBinding(QUOTE_COMMAND_ID, KeyboardPrefs.quote);
+    KeyBindingManager.addBinding(CODEBLOCK_COMMAND_ID, KeyboardPrefs.codeblock);
+    KeyBindingManager.addBinding(PARAGRAPH_COMMAND_ID, KeyboardPrefs.paragraph);
+    KeyBindingManager.addBinding(REFLOW_COMMAND_ID, KeyboardPrefs.reflow);
+
+
+
+    ExtensionUtils.loadStyleSheet(module, "styles/styles.css");
+    ExtensionUtils.loadStyleSheet(module, "styles/octicons.css");
+    barShouldShow = true;
+
+    activeEditorChangeHandler(null, EditorManager.getActiveEditor(), null);
+    EditorManager.on("activeEditorChange", activeEditorChangeHandler);
+
+    var editor = $(" #editor-holder");
+    editor.css("display", "block");
+    //TODO: update the code in workspace manager to handle code editor height adjustment in initial rendering
+    WorkspaceManager.recomputeLayout(true);
+    //editor height gets adjusted after any operation including toggling, resizing
+    editor.css("height","80%");     //temporary height fix
```

Comment:
I have tried figuring out the problem with initial code rendering issue. I have investigated that `WorkspaceManager.recomputeLayout()` is responsible for recomputing the layout once a widget is added. But, calling within main does not seem to find the fix. Hence, to support functioning temporarily , I have provided a height 80% fix (temporarily).

Reply:
Its quite interesting to see that the function works perfectly oce we toggle the tabs, resize the frame etc ... or any action performed. The problem is only with  initial rendering. Can we identify it as an issue ?